src/Session.py

```diff
@@ -59,6 +60,7 @@ def openSession(self, sessionID, server):
         """
         self.__sessionID = sessionID
         self.__server = server
+        self.__secure = secure
```

Comment:
@Pazze12 Fehlt `secure` hier nicht in den übergebenen Argumenten?
Weiterhin sollte in der Funktion  `closeSession` die Eigenschaft `__secure` wieder zurückgesetzt werden.

Reply:
@MasterZydra secure wird nicht genutzt

Reply:
```suggestion
        self.__secure = secure
```

muss oben definiert werden. z.B.
```
def openSession(self, sessionID, server, secure):
```
oder global

Reply:
... oder als optionales Argument mit None als Beispielwert.
```
def openSession(self, sessionID, server, secure = None):
```

Reply:
Ich muss mich berichtigen. None als Standardwert für Secure ist schlecht. Besser ist das was du im __init__ schon gemacht hast, nämlich ''.  Wo wird secure gesetzt, außer im __init__?  Ohne zusätzlichen Übergabeparameter dürfte bei openSession eine Exception geworfen werden. So wie ich das gesehen habe wird es in der HTTP Connection benötigt und kann entweder den Wert '' oder 's' annehmen. Für http bzw. https. Ist das richtig?
Gibt es den http Fall überhaupt noch? Ich dachte Upjers hat auf https umgestellt. Daher stellt sich für mich sogar die Frage, ob das secure überhaupt benötigt wird.

Reply:
nicht sicher brauchte es aber für den portallogin

Reply:
@Pazze12 @MrFlamez Wie wäre es in HTTPConnection ein Flag für HTTPS einzubauen? Die Session könnte dieses setzen. Damit kann der Typ hier auf einen optionalen Bool geändert werden.

Reply:
In HTTPConnection könnte das `http` in der URL durch die Funktion `self.getProtocol()` ersetzt werden. Dort wird je nach Flag `http` oder `https` zurückgegeben.

Reply:
Die Session hat damit nichts zu tun. Ich hätte das, wenn dann, beim Erzeugen des Objektes der HTTP Connection gemacht. Ich hätte das vermutlich komplett auf HTTPS umgebaut. Upjers wird sicher keinen Schritt zurück machen.